### ISSUE TITLE
Toggling Fingerprint in Android settings causes issues

### ISSUE THREAD

**Issue by ntuanb** [mjwheatley/cordova-plugin-android-fingerprint-auth#15]
Steps to replicate:
- Activate the fingerprint in the Android OS. (FingerprintAuth.isAvailable() returns true)
- Call FingerprintAuth.show() and works successfully. (displays fingerprint popup)
- Deactivate the fingerprint in the Android OS. (FingerprintAuth.isAvailable() returns false)
- Call FingerprintAuth.show() and fails as expected.
- Reactivate the fingerprint in the Android OS (FingerprintAuth.isAvailable() returns true)
- Call FingerprintAuth.show() returns an error message "Failed to init Cipher and backup disabled."

Note: the clientId and clientSecret remained unchanged for all steps.

**Comment by ntuanb** [comment]
Log:

FingerprintAuth: Failed to init Cipher: InvalidKeyException: android.security.keystore.KeyPermanentlyInvalidatedException: Key permanently invalidated

**Comment by ntuanb** [comment]
Didn't realise we had to enable the backup when a new fingerprint is enrolled. 

Perhaps the explanation of the backup functionality should be clearer?

**Comment by mjwheatley** [comment]
I was going to say it looked like you were setting the disableBackup parameter to true if you were getting the "Failed to init Cipher and backup disabled." message.
I did design the plugin to invalidate the key in the keystore for the app if the enrolled fingerprints on the device change as a security feature, thus requiring a user to re-authenticate using the backup. https://developer.android.com/reference/android/security/keystore/KeyPermanentlyInvalidatedException.html
```
mKeyGenerator.init(new KeyGenParameterSpec.Builder(mClientId,
                    KeyProperties.PURPOSE_ENCRYPT |
                            KeyProperties.PURPOSE_DECRYPT)
                    .setBlockModes(KeyProperties.BLOCK_MODE_CBC)
                            // Require the user to authenticate with a fingerprint to authorize every use
                            // of the key
                    .setUserAuthenticationRequired(true)
                    .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_PKCS7)
                    .build());
```
I could potentially create a plugin parameter to pass to the setUserAuthenticationRequired() so the developer could decide if they want the user's key to be invalidated or not under these circumstances.  Is that something you would be interested in?

**Comment by ntuanb** [comment]
Sounds great if that could be added in as an option.

**Comment by malwatte** [comment]
@mjwheatley ,
Any updates on the new "setUserAuthenticationRequired" parameter.
I want it too :)
Thanks

**Comment by mjwheatley** [comment]
added userAuthRequired parameter and published v0.1.0

Keep in mind that if it was true, you will have to use the backup authentication 1 time to create a new key with userAuthRequired set to false.
